app/src/main/java/com/example/androidtraining/TellerOnlineRepository.kt

```diff
@@ -25,22 +26,20 @@ class TellerOnlineRepository(private val db: GitHubRepoDataBase, private val ser
 
     override fun fetchFreshCache(requirements: GetReposRequirement): Single<FetchResponse<GitHubRepoList>> {
         return service.getRepos(requirements.dayInformation.getYesterday())
-            .map {response -> val fetchResponse: FetchResponse<GitHubRepoList> =
-                if (response.isError){
-                    val errorResult = when(response.error()!!){
-                        is IOException -> {FetchResponse.fail<GitHubRepoList>(NetworkError())}
-                        else -> {FetchResponse.fail(UnhandledError())}
+            .map { result ->
+                val processedResponse = responseProcessor.process(result) { code, response, errorBody, jsonAdapter ->
```

Comment:
`responseProcessor.process()` takes an optional 2nd paramter which is a lambda where you can handle http errors manually. You would do that where an endpoint throws unique errors that other endpoints don't such as this endpoint. 

Notice how we are using the moshi jsonAdapter to manually parse the json response body into an instance of the UserEnteredBadDataResponseError data class.